### ISSUE TITLE
false redefined-outer-name when using `del`

### ISSUE THREAD

**Issue by vapier** [PyCQA/pylint#841]
simple test code:
```
path = __file__
del path
def some_func_of_mine(path):
    print(path)
```

testing pylint-1.5.4 shows:
```
$ pylint test.py
W:  3,22: Redefining name 'path' from outer scope (line 1) (redefined-outer-name)
```

**Comment by PCManticore** [comment]
We don't understand deletions of values yet.

**Comment by dumblob** [comment]
This would be great addition. It's one of the most frequent errors we always get.

I didn't look at the internals, but would it make sense to look at how *mypy* solves this (*mypy* understands `del` statements and does not produce any false positivess)?

**Comment by clavedeluna** [comment]
I'm looking at how to approach this. [this](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/variables.py#L1219) is the code near where "redefined-outer-name" is about to be emitted.

In the orig example in this issue

```
path = __file__
del path
def some_func_of_mine(path):
    print(path)
```

`globs` is

```
(Pdb) globs
{'path': [<AssignName.path l.1 at 0x10ad69f40>, <DelName.path l.2 at 0x10ade0070>], 'some_func_of_mine': [<FunctionDef.some_func_of_mine l.3 at 0x10ade0040>]}
```

Seems like we could use this information, particularly the fact that `DelName.path` exists, and short-circuit so that `Redefining name 'path' from outer scope (line 1) (redefined-outer-name)` isn't part of the output.

Thoughts?

**Comment by Pierre-Sassoulas** [comment]
This seems dependent on https://github.com/PyCQA/pylint/issues/4795, doing it in each possible checker will be very hard to maintain and do right.

**Comment by clavedeluna** [comment]
> This seems dependent on #4795, doing it in each possible checker will be very hard to maintain and do right.

Sounds good. Then the minor label isn't applicable, correct?

**Comment by Pierre-Sassoulas** [comment]
minor means "polishing" i.e. low priority, not "easy to do " :)
